subt_ign/src/GameLogicPlugin.cc

```diff
@@ -663,6 +684,17 @@ void GameLogicPlugin::Configure(const ignition::gazebo::Entity & /*_entity*/,
       this->dataPtr->rosRegionEventPub =
         this->dataPtr->rosnode->advertise<subt_ros::RegionEvent>(
             "region_event", 100);
+
+      // Setup a ros bag recorder.
+      rosbag::RecorderOptions recorderOptions;
+      recorderOptions.append_date=false;
+      recorderOptions.prefix="/tmp/ign/logs/cloudsim";
+      recorderOptions.regex=true;
+      recorderOptions.topics.push_back("/subt/.*");
+
+      this->dataPtr->rosRecorder.reset(new rosbag::Recorder(recorderOptions));
+      this->dataPtr->bagThread.reset(new std::thread(
+        &GameLogicPluginPrivate::RosBag, this->dataPtr.get()));
```

Comment:
```suggestion
      this->dataPtr->rosRecorder = std::make_unique<rosbag::Recorder>(recorderOptions);
      this->dataPtr->bagThread = std::make_unique<std::thread>([&](){
          this->dataPtr->rosRecorder->run();
      });
```

I think you can save yourself a member function here.

Reply:
I also did this in  bc03832 since I was in there.